x-pack/plugins/infra/server/routes/metrics_explorer/lib/convert_request_to_metrics_api_options.ts

```diff
@@ -9,13 +9,16 @@ import {
   MetricsAPIRequest,
   MetricsExplorerRequestBody,
   afterKeyObjectRT,
+  MetricsAPIMetric,
 } from '../../../../common/http_api';
 import { convertMetricToMetricsAPIMetric } from './convert_metric_to_metrics_api_metric';
 
 export const convertRequestToMetricsAPIOptions = (
   options: MetricsExplorerRequestBody
 ): MetricsAPIRequest => {
-  const metrics = options.metrics.map(convertMetricToMetricsAPIMetric);
+  const metrics = options.metrics
+    .map(convertMetricToMetricsAPIMetric)
+    .filter((m) => m) as MetricsAPIMetric[];
```

Comment:
How about expressing this without an explicit type cast?

```suggestion
    .filter(<M>(m: M): m is NonNullable<M> => !!m);
```